glyphs-reader/src/font.rs

```diff
@@ -1702,31 +1658,136 @@ impl TryFrom<RawFont> for Font {
+        Regex::new(r"(?mi)^(?P<prefix>\s*glyphname\s*=\s*)(?P<value>(?:infinity|inf|nan));\s*$")
+            .unwrap();
+    let s = unicode_re.replace_all(&s, r#"$prefix"$value";"#);
+    let s = glyphname_re.replace_all(&s, r#"$prefix"$value";"#);
+    s.into_owned()
+}
+
 impl Font {
     pub fn load(glyphs_file: &path::Path) -> Result<Font, Error> {
-        debug!("Read {glyphs_file:?}");
-        let raw_content = fs::read_to_string(glyphs_file).map_err(Error::IoError)?;
+        if glyphs_file.is_dir() {
+            return Font::load_package(glyphs_file);
+        }
 
-        // Glyphs has a wide variety of unicode definitions, not all of them parser friendly
-        // Make unicode always a string, without any wrapping () so we can parse as csv, radix based on format version
-        let re =
-            Regex::new(r"(?m)^(?P<prefix>\s*unicode\s*=\s*)[(]?(?P<value>[0-9a-zA-Z,]+)[)]?;\s*$")
-                .unwrap();
-        let raw_content = re.replace_all(&raw_content, r#"$prefix"$value";"#);
+        debug!("Read {glyphs_file:?}");
+        let raw_content =
+            preprocess_unparsed_plist(fs::read_to_string(glyphs_file).map_err(Error::IoError)?);
 
-        let mut raw_content = Plist::parse(&raw_content)
+        let raw_content = Plist::parse(&raw_content)
             .map_err(|e| Error::ParseError(glyphs_file.to_path_buf(), format!("{e:#?}")))?;
 
-        // Fix any issues with the raw plist
-        let Plist::Dictionary(ref mut root_dict) = raw_content else {
+        let raw_font = RawFont::from_plist(raw_content);
+        raw_font.try_into()
+    }
+
+    pub fn load_package(glyphs_package: &path::Path) -> Result<Font, Error> {
+        if !glyphs_package.is_dir() {
+            return Err(Error::NotAGlyphsPackage(glyphs_package.to_path_buf()));
+        }
+        debug!("Read {glyphs_package:?}");
+
+        let fontinfo_file = glyphs_package.join("fontinfo.plist");
+        let fontinfo_data = fs::read_to_string(&fontinfo_file).map_err(Error::IoError)?;
+        let mut font_plist = Plist::parse(&fontinfo_data)
+            .map_err(|e| Error::ParseError(fontinfo_file.to_path_buf(), format!("{e:#?}")))?;
+
+        let Plist::Dictionary(ref mut root_dict) = font_plist else {
             return Err(Error::ParseError(
-                glyphs_file.to_path_buf(),
+                fontinfo_file.to_path_buf(),
                 "Root must be a dict".to_string(),
             ));
         };
-        fix_glyphs_named_infinity(glyphs_file, root_dict)?;
 
-        let raw_font = RawFont::from_plist(raw_content);
+        let mut glyphs: HashMap<String, Plist> = HashMap::new();
+        let glyphs_dir = glyphs_package.join("glyphs");
+        if glyphs_dir.is_dir() {
+            for entry in fs::read_dir(glyphs_dir).map_err(Error::IoError)? {
+                let entry = entry.map_err(Error::IoError)?;
+                let name = entry.file_name();
+                if name.to_string_lossy().ends_with(".glyph") {
```

Comment:
since we need the path anyway below we can skip needing to separately get the file name, and do something like,

```suggestion
                let path = entry.path();
                if path.extension() == Some(OsStr::new("glyph")) {
                // ...
                // reuse path below
```